drivers/rn2xx3/rn2xx3_internal.c

```diff
@@ -0,0 +1,352 @@
+/*
+ * Copyright (C) 2017 Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     driver_rn2xx3
+ * @{
+ *
+ * @file
+ * @brief       Internal driver implementation for the RN2483/RN2903 devices
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include "rn2xx3_internal.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#define RESP_TIMEOUT_SEC            (5U)
+
+static void _terminate_command(rn2xx3_t *dev)
+{
+    /* \r\n is the command closing sequence */
+    uart_write(dev->p.uart, (uint8_t*)"\r\n", 2);
```

Comment:
About all uses of uart_write in this module write null terminated strings to the output. How about introducing a helper function. (Maybe I already suggested this before, but I forgot if I did)

    static void _my_uart_write(rn2xx3_t *dev, const char *str)
    {
        size_t len = strlen(str);
        if (len) {
            uart_write(dev->p.uart, (uint8_t*)str, len);
        }
    }

And in the rest of the module it can be as simple as

    _my_uart_write(dev, "\r\n");
    _my_uart_write(dev, dev->cmd_buf);